src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -10027,39 +10027,9 @@ public static void M()
 
             var expectedDiagnostics = new[]
             {
-                // (7,26): error CS1513: } expected
-                //         var b = a with { [0] = 20 };
-                Diagnostic(ErrorCode.ERR_RbraceExpected, "[").WithLocation(7, 26),
-                // (7,26): error CS1002: ; expected
-                //         var b = a with { [0] = 20 };
-                Diagnostic(ErrorCode.ERR_SemicolonExpected, "[").WithLocation(7, 26),
-                // (7,26): error CS7014: Attributes are not valid in this context.
-                //         var b = a with { [0] = 20 };
-                Diagnostic(ErrorCode.ERR_AttributesNotAllowed, "[").WithLocation(7, 26),
-                // (7,27): error CS1001: Identifier expected
-                //         var b = a with { [0] = 20 };
-                Diagnostic(ErrorCode.ERR_IdentifierExpected, "0").WithLocation(7, 27),
-                // (7,27): error CS1003: Syntax error, ']' expected
-                //         var b = a with { [0] = 20 };
-                Diagnostic(ErrorCode.ERR_SyntaxError, "0").WithArguments("]").WithLocation(7, 27),
-                // (7,28): error CS1002: ; expected
-                //         var b = a with { [0] = 20 };
-                Diagnostic(ErrorCode.ERR_SemicolonExpected, "]").WithLocation(7, 28),
-                // (7,28): error CS1513: } expected
-                //         var b = a with { [0] = 20 };
-                Diagnostic(ErrorCode.ERR_RbraceExpected, "]").WithLocation(7, 28),
-                // (7,30): error CS1525: Invalid expression term '='
-                //         var b = a with { [0] = 20 };
-                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "=").WithArguments("=").WithLocation(7, 30),
-                // (7,35): error CS1002: ; expected
-                //         var b = a with { [0] = 20 };
-                Diagnostic(ErrorCode.ERR_SemicolonExpected, "}").WithLocation(7, 35),
-                // (7,36): error CS1597: Semicolon after method or accessor block is not valid
+                // (7,26): error CS0747: Invalid initializer member declarator
                 //         var b = a with { [0] = 20 };
```

Comment:
>with { [0] = 20 } [](http://example.com/codeflow?start=37&length=17)

Should this be parse error?

Reply:
Looking.  This may actually be a bug.

Reply:
So, currently, ParseWithExpression allows *all* expressions within the initializer.  Then, in BindObjectInitializerMember we bail out if it's not an expr we support.  And then lower we report that it's an invalid initializer.  So i think this is all working as intended (we do the same for any other aribtrary expressions here).